hearth/lib/hearth/middleware/send.rb

```diff
@@ -86,30 +91,50 @@ def send_request(context, output)
       end
 
       def apply_stub(stub, input, context, output)
+        # require 'byebug'
+        # byebug
         case stub
         when Proc
           stub = stub.call(input, context)
           apply_stub(stub, input, context, output) if stub
         when Exception, ApiError
           output.error = stub
         when Hash
-          @stub_class.stub(
-            context.response,
-            stub: @params_class.build(
-              stub,
-              context: 'stub'
+          if stub.key?(:error)
```

Comment:
I'd still argue for either:
1) Check here for something like `stub.key?(:error) and stub[:error].is_a?(Class)`.  And then avoid the data key.  (since an output w/ a class value isn't valid - it avoids any chance of an api overlapping with this).
2) Creating a `ResponseError` class and adding a `when ResponseError` to this `case`.

I just think errors are a less common stubbing case and I want to avoid friction in upgrading from V3 to V4 and avoid interface changes where its reasonable....

Reply:
We've already made a lot of friction such as Config object, namespace, return Output instead of data, etc. I agree with the sentiment. I actually think since hash is ambiguous, using data/error key is the correct call. It also allows us to maybe stub metadata in the future? The other thing I thought of was maybe just an array is assumed to be an error but that is kinda wonky.